### ISSUE TITLE
Model Not Converge on NYU40 Dataset

### ISSUE THREAD

**Issue by riven314** [montrealrobotics/ENetDepth#3]
Thanks for your prompt reply with regard to our previous issue. 
I tried to your ENet with NYU40 dataset, but I found that the model does not converge at the end, leading to a worst mean IoU as shown:
![6](https://user-images.githubusercontent.com/21143399/65828423-509ab500-e2cd-11e9-81eb-7e4c7ef976b6.jpg)

The following is the reported training loss at the end of few epochs:

![2](https://user-images.githubusercontent.com/21143399/65828425-5b554a00-e2cd-11e9-824e-c71a197b0303.jpg)

The configuration I used is pretty similar to the one you used on ScanNet. Did you train your ENet on NYU40 before and was its performance similar to mine?

**Comment by krrish94** [comment]
Not an "issue".

**Comment by krrish94** [comment]
I'm assuming this is a casual case of hyperparameter tuning. I don't have a trained model at hand for NYU40, but this should work provided all other hyperparameters are correctly set (eg. learning rates/batch size/beta). I'd recommend you to try simple memorization experiments first (eg. overfitting to a minibatch, overfitting to 3-4 minibatches, etc.) before moving on to the full dataset.

In general, I found the performance of the RGB-D variants to be much better than that of the RGB variants.

**Comment by riven314** [comment]
Thanks for your suggestion. Would you mind me posting other questions here? I think they don't belong to any issue:
I have processed the ScanNet data with your data pre-processing script. I downloaded the scenes that you specified in train.txt and test.txt. I later encountered an error in inference.py (line 103: loss = self.criterion(outputs, labels)). 
I found that it is because the variable "output" has a different dimension than "labels". While "output" has dimension (# batch size, 21, height, width), "labels" has dimension (# batch size, height, width). I intended to do one-hot encoding on "labels" so as to expand the number of channels, but I noticed that "labels" has some integers greater than 20 (it is expected to be 0-20 as there are 21 classes). 
Did you encounter similar problem and how do you handle those unwanted class id?

**Comment by krrish94** [comment]
I'm not entirely sure, but here's what my preliminary guess is. The codebase supports two datasets `NYU40` and `ScanNet`. There could either be
1. A mix-up in the processing script that's mistakenly assigning NYU40 labels instead of ScanNet labels
2. An incorrect command-line argument supplied to the script when you're calling it.

I have a hunch, it's the latter. But in case it's a bug with the repo, feel free to raise an issue and I'll fix it asap.
